src/fheroes2/agg/agg_image.cpp

```diff
@@ -1794,6 +1845,66 @@ namespace fheroes2
 
                 break;
             }
+            case ICN::BUTTON_VERTICAL_DISMISS: {
+                _icnVsSprite[id].resize( 2 );
+
+                if ( useOriginalResources() ) {
+                    // The original DISMISS button has a broken transform layer and thinner pressed state than released state,
+                    // so we use our empty vertical button as a template
+                    _icnVsSprite[id][0] = GetICN( ICN::EMPTY_VERTICAL_GOOD_BUTTON, 0 );
+                    _icnVsSprite[id][1] = GetICN( ICN::EMPTY_VERTICAL_GOOD_BUTTON, 1 );
+
+                    // Copy the DISMISS text back from the original button
+                    const Sprite & originalReleased = GetICN( ICN::HSBTNS, 0 );
+                    const Sprite & originalPressed = GetICN( ICN::HSBTNS, 1 );
+
+                    Copy( originalReleased, 9, 2, _icnVsSprite[id][0], 4, 2, 21, 112 );
+                    Copy( originalPressed, 9, 5, _icnVsSprite[id][1], 3, 5, 19, 111 );
+
+                    fheroes2::makeTransparentBackground( _icnVsSprite[id][0], _icnVsSprite[id][1], ICN::REDBAK_SMALL_VERTICAL );
+
+                    break;
+                }
+
+                // We need to temporarily remove the letter specific X offsets in the font because if not the letters will
+                // be off-centered when we are displaying one letter per line
+                const ButtonFontRestorer fontReleased( _icnVsSprite[ICN::BUTTON_GOOD_FONT_RELEASED], { -1, 0 } );
+                const ButtonFontRestorer fontPressed( _icnVsSprite[ICN::BUTTON_GOOD_FONT_PRESSED], { -1, 0 } );
+                getTextAdaptedButton( _icnVsSprite[id][0], _icnVsSprite[id][1], gettext_noop( "D\nI\nS\nM\nI\nS\nS" ), ICN::EMPTY_VERTICAL_GOOD_BUTTON,
+                                      ICN::REDBAK_SMALL_VERTICAL );
+
+                break;
+            }
+            case ICN::BUTTON_VERTICAL_EXIT: {
+                _icnVsSprite[id].resize( 2 );
+
+                if ( useOriginalResources() ) {
+                    // The original EXIT button has a broken transform layer and we need to remove the shadows,
+                    // so we use our empty vertical button as a template
+                    _icnVsSprite[id][0] = GetICN( ICN::EMPTY_VERTICAL_GOOD_BUTTON, 0 );
+                    _icnVsSprite[id][1] = GetICN( ICN::EMPTY_VERTICAL_GOOD_BUTTON, 1 );
+
+                    // Copy the EXIT text back from the original button
+                    const Sprite & originalReleased = GetICN( ICN::HSBTNS, 0 );
+                    const Sprite & originalPressed = GetICN( ICN::HSBTNS, 1 );
+
+                    Copy( originalReleased, 9, 2, _icnVsSprite[id][0], 4, 2, 21, 112 );
+                    Copy( originalPressed, 9, 5, _icnVsSprite[id][1], 3, 5, 19, 111 );
```

Comment:
For original EXIT button we need to use 2 and 3 image index and use different X offsets in these images when performing `Copy()`

```suggestion
                    const Sprite & originalReleased = GetICN( ICN::HSBTNS, 2 );
                    const Sprite & originalPressed = GetICN( ICN::HSBTNS, 3 );

                    Copy( originalReleased, 4, 2, _icnVsSprite[id][0], 4, 2, 21, 112 );
                    Copy( originalPressed, 3, 5, _icnVsSprite[id][1], 3, 5, 19, 111 );
```